src/USER-MESONT/pair_mesont_tpm.cpp

```diff
@@ -508,58 +520,58 @@ void PairMESONTTPM::compute(int eflag, int vflag){
     error->all(FLERR, err.str().c_str());
   }
 
-  // set per atom values and accumulators
-  // reallocate per-atom arrays if necessary
-  if (atom->nmax > maxeatom) {
-    maxeatom = atom->nmax;
-    memory->destroy(eatom);
-    memory->create(eatom,comm->nthreads*maxeatom,"pair:eatom");
-    memory->destroy(eatom_s);
-    memory->create(eatom_s,comm->nthreads*maxeatom,"pair:eatom_s");
-    memory->destroy(eatom_b);
-    memory->create(eatom_b,comm->nthreads*maxeatom,"pair:eatom_b");
-    memory->destroy(eatom_t);
-    memory->create(eatom_t,comm->nthreads*maxeatom,"pair:eatom_t");
+  //convert from sorted representation
+  for (int i = 0; i < nall; i++){
+      int idx = ntlist.get_idx(i);
+      for (int j = 0; j < 3; j++) f[idx][j] += f_sort[3*i+j];
+      buckling[idx] = b_sort[i];
   }
-
-  if (atom->nmax > maxvatom) {
-    maxvatom = atom->nmax;
-    memory->destroy(vatom);
-    memory->create(vatom,comm->nthreads*maxvatom,6,"pair:vatom");
+  if(eflag){
```

Comment:
this is still not good enough. eflag is nonzero if either eflag_atom or eflag_global is set, but you can use eatom *only* if eflag_atom is set (and I would assume the same is true for the other eatom_* variants, otherwise only global accumulators are available.

Reply:
Thank you for pointing it out, I already realized it and wanted to separate the evaluation of per atom and global energies.